lib/ansible/modules/windows/win_get_url.ps1

```diff
@@ -19,120 +20,186 @@
 # WANT_JSON
 # POWERSHELL_COMMON
 
+
+Function CheckModified-File($url, $dest, $headers, $credentials, $timeout, $use_proxy, $proxy) {
```

Comment:
I see my misunderstanding, the only thing I am concerned about is scenario 1 and 3. I could be wrong but I am under the assumption that if nothing is set (current scenario today) it will not use any proxies even if one is set on the system. If you wish to use the system proxy then you would have to set and assign `$proxy` to that property.

```
$proxy = [System.Net.WebRequest]::GetSystemWebProxy()
$proxy.Credentials = [System.Net.CredentialCache]::DefaultCredentials
```
Unfortunately I don't have an environment I can test this assumption in but I believe in this case scenario 3 isn't catered for in your code.